src/types.rs

```diff
@@ -477,10 +479,11 @@ impl Rewrite for ast::PolyTraitRef {
     fn rewrite(&self, context: &RewriteContext, width: usize, offset: Indent) -> Option<String> {
         if !self.bound_lifetimes.is_empty() {
             let lifetime_str = try_opt!(self.bound_lifetimes
-                    .iter()
-                    .map(|lt| lt.rewrite(context, width, offset))
-                    .collect::<Option<Vec<_>>>())
-                .join(", ");
+                .iter()
+                .map(|lt| lt.rewrite(context, width, offset))
+                .intersperse(Some(", ".to_string()))
+                .collect::<Option<String>>());
```

Comment:
Update to this, they're not actually needed for each one, so removing them now to clean this up a little.